tests/integration/helm_otc_metadata_installation_test.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal"
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/ctxopts"
+	"github.com/SumoLogic/sumologic-kubernetes-collection/tests/integration/internal/stepfuncs"
+)
+
+func Test_Helm_OT_Metadata(t *testing.T) {
+	var (
+		now            = time.Now()
+		namespace      = generateNamespaceName(now)
+		releaseName    = generateReleaseName(now)
+		valuesFilePath = "values/values_otc_metadata.yaml"
+
+		tickDuration = time.Second
+		waitDuration = time.Minute
+	)
+
+	feat := features.New("installation").
+		// Setup
+		Setup(stepfuncs.SetKubectlNamespaceOpt(namespace)).
+		Setup(stepfuncs.KubectlApplyFOpt(internal.YamlPathReceiverMock, "receiver-mock")).
+		Setup(stepfuncs.SetHelmOptionsOpt(valuesFilePath)).
+		Setup(stepfuncs.HelmDependencyUpdateOpt(internal.HelmSumoLogicChartAbsPath)).
+		Setup(stepfuncs.HelmInstallOpt(internal.HelmSumoLogicChartAbsPath, releaseName)).
+		// Teardown
+		Teardown(stepfuncs.PrintClusterStateOpt()).
+		Teardown(stepfuncs.HelmDeleteOpt(releaseName)).
+		Teardown(stepfuncs.KubectlDeleteNamespaceOpt(namespace)).
+		Teardown(stepfuncs.KubectlDeleteFOpt(internal.YamlPathReceiverMock, "receiver-mock")).
+		// Assess
+		Assess("sumologic secret is created",
+			func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+				k8s.WaitUntilSecretAvailable(t, ctxopts.KubectlOptions(ctx), "sumologic", 60, tickDuration)
+				secret := k8s.GetSecret(t, ctxopts.KubectlOptions(ctx), "sumologic")
+				require.Len(t, secret.Data, 10)
+				return ctx
+			}).
+		Assess("3 otelcol logs pods are created, running and ready",
+			func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+				require.Eventually(t, func() bool {
+					pods := k8s.ListPods(t, ctxopts.KubectlOptions(ctx), v1.ListOptions{
+						LabelSelector: fmt.Sprintf("app=%s-sumologic-otelcol-logs", releaseName),
+						FieldSelector: "status.phase=Running",
+					})
+
+					if len(pods) != 3 {
+						return false
+					}
```

Comment:
I think we should move the magic numbers and strings, like the pod count or the app labels, into the variable definitions up top. In all honesty, we should either read these from the generated yaml or explicitly template them into the values.yaml we use for tests, but making them more explicit should be good enough for now.

This applies to all the labels and pod counts in this test.

Reply:
I get the idea and we can perhaps reorganize/templatize this at some point but one thing I wouldn't change which is to not read the (default) values from the yaml files but to assert on them (expecting what's the actual default) if that makes sense.